### PR TITLE
Compiler fixes 4

### PR DESCRIPTION
**Pull request by evertvorster**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [x] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues
#257 
#253 
#249 
#248 

Purpose:
- What is this pull request trying to do?
Clear some compiler warnings without changing the game in any way

- What release is this for?
Any

- Is there a project or milestone we should apply this to?
Nope.

### CODE REVIEW

**Comment by evertvorster** [comment]
I'll add some more compiler fixes to this branch as I go along, one commit per issue to keep 'em separated.

**Comment by evertvorster** [comment]
Have at it, tell me where I have gone wrong in my fixes!

File: engine/src/gldrv/gl_light_pick.cpp [hunk 1]
```diff
@@ -30,10 +30,6 @@ struct light_key
         intensity_key = inte;
     }
 };
-static bool operator<( light_key tmp1, light_key tmp2 )
-{
-    return tmp1.intensity_key < tmp2.intensity_key;
-}
```
Comment (stephengtuggy) [inline engine/src/gldrv/gl_light_pick.cpp:36]: It looks like, while this `operator <` technically wasn't **used** in the Vega Strike code, it was nevertheless necessary for implementing the template `less`. We'll need to put it back. And maybe add a `#pragma warn...` to disable the warning on it.

Other than that, your fixes look really good.

**Review by stephengtuggy — CHANGES_REQUESTED** [review]
Looks like we need to put `operator<` back. Other than that, things are looking good.

File: engine/src/gldrv/gl_light_pick.cpp (hunk 1 above)
Comment (evertvorster) [reply engine/src/gldrv/gl_light_pick.cpp]: Thanks for the vote of confidence, as well as the hint on how to make gcc ignore a warning. The code has been modified as per your instructions.

**Review by stephengtuggy — APPROVED** [review]
Looks good to me!

**Pull request merged**